Assets/Scripts/SceneLoader.cs

```diff
@@ -1,113 +1,126 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
+using System.IO;
 using UnityEngine;
 using UnityEngine.SceneManagement;
 using TMPro;
 using UnityEngine.UI;
+using System;
 
 public class SceneLoader : MonoBehaviour
 {
     public GameObject LevelSelector;
     public GameObject LevelContent;
-    private string starprogress = "";
-    // Start is called before the first frame update
-    void Start()
+    private string playerMinSaves = "0;-1;-1;-1;-1;-1;-1;-1;-1;-1";
+
+    // Stores max number of saves required per star per level
+    private List<List<int>> savesRequired = new List<List<int>>();
+
+    // Read information about max number of saves required per star per level
+    private void ParseStarInfo()
     {
+        string text = File.ReadAllText("./Assets/Resources/starinfo.txt");
+        string[] lines = text.Split('\n');
+
+        savesRequired = new List<List<int>>();
+        foreach (string line in lines)
+        {
+            if (line == "") break;
+            string[] info = line.Split(' '); // level number, 1 star, 2 star, 3 star
+            List<int> saves = new List<int> (new int[] { int.Parse(info[1]), int.Parse(info[2]), int.Parse(info[3]) } );
+            savesRequired.Add(saves);
+        }
+    }
 
-        if (PlayerPrefs.HasKey("starprogress"))
+    // Retrieve data about player's smallest ever no. of saves per level
+    // -1 indicates the level is locked, 0 indicates the level is unlocked but not attempted
+    private void GetPlayerProgress()
+    {
+        if (PlayerPrefs.HasKey("playerMinSaves"))
         {
-            starprogress = PlayerPrefs.GetString("starprogress");
+            playerMinSaves = PlayerPrefs.GetString("playerMinSaves");
         }
         else
         {
-            starprogress = "0;-1;-1;-1;-1;-1;-1;-1;-1;-1";
-            PlayerPrefs.SetString("starprogress", starprogress);
+            PlayerPrefs.SetString("playerMinSaves", playerMinSaves);
         }
+    }
```

Comment:
Create helper function for reinitialising playerMinSaves to the correct number of -1's depending on the num of levels (from starinfo file).
If len(existing playerMinSaves from player prefs) != num of levels, reset all progress by reinitialising playerMinSaves.